AK/DistinctNumeric.h

```diff
@@ -38,9 +38,6 @@ namespace AK {
  * I intentionally decided against overloading `&a` because these shall remain
  * numeric types.
  *
- * The C++20 `operator<=>` would require, among other things `std::weak_equality`.
```

Comment:
This comment looks very strange given that you can use any type that has the states "zero", "positive" and "negative", e.g. `int`. Did the commenter even read cppreference?

Reply:
Integral types, like we use as `m_value` for this type, always return `std::strong_ordering` from their own `operator<=>`. Until I saw Andreas use it for the new AK::String I also held the assumption from the comment.

Changing it to 
```
constexpr auto operator<=>(Foo const&) const = default;
```
complains: Please include `<compare>`

Changing it to 
```
constexpr auto operator<=>(Foo const& other) const
{
    return m_value <=> other.m_value;
}
```
complains: Please include `<compare>`.

I ran into a bug with clang and our overloaded `operator<` from AK::Time.h that prevented operator<=> on that type from compiling until I removed those. I suspect that the global `operator<` overloads in AK/Checked.h will have a similar effect. 

So, no, I don't find it that strange that the commenter believed this to be true two years ago. It's likely that cppreference has updated the guidance to be more obvious since then too.